src/NuGet.Core/NuGet.Protocol/Model/RegistrationIndexResult.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Protocol.Model
+{
+    internal class RegistrationIndexResult
```

Comment:
This looks very specific to one method call, rather than a generically useful. I suggest deleting this class, and using a ValueTuple instead. This also has a slight perf improvement, because ValueTuple is a struct, meaning it won't cause an allocation for an object that will later need to be garbage collected, just to wrap this simple return value.

```cs
Task<(RegistrationIndex Index, HttpSourceCacheContext CacheContext)> MethodAsync()
{
  // code goes here
  return (index, cacheContext);
}

(index, cacheContext) = await MethodAsync();
```

Reply:
@zivkan I noticed GlobalSuppression.cs file doesn't work if method returns ValueTuple.
Suppressions still show even though I already suppressed it.
Do you work around it?

Reply:
Maybe you could use `ValueTuple<RegistrationIndex, HttpSourceCacheContext>` in the method signature, that will create a name that can be used in the global supressions file.

Reply:
You can insert suppression attribute locally, which does not need member signature, or `\#pragma`

Reply:
Looks working. Once unit testing on cloud passes then I'll merge my PR.